tests/python_test.py

```diff
@@ -512,10 +516,21 @@ def raises_sanlock_errno(expected_errno=errno.ECONNREFUSED):
 
 @pytest.mark.parametrize("name", LOCKSPACE_OR_RESOURCE_NAMES)
 @pytest.mark.parametrize("filename,encoding", FILE_NAMES)
-def test_rem_lockspace_parse_args(no_sanlock_daemon, name, filename, encoding):
+@pytest.mark.parametrize("flags", [
+        pytest.param({"wait": True, "async": False},
+            marks=[
+                    pytest.mark.xfail(six.PY2, raises=RuntimeError, reason="async and wait flags must be mutually exclusive"),
+                    pytest.mark.xfail(six.PY3, raises=TypeError, reason="async flag is deprecated by wait"),
+                ]),
```

Comment:
Both cases should have negative test:

- using async= on py3 raises TypeError
- using both wait= and async= on python2 raises RuntimeError

We don't use xfail as a way to test negative flows. xfail is a way to keep the tests pass before we can fix the real issue.